Utilities/rANS/include/rANS/Encoder.h

```diff
@@ -88,31 +90,53 @@ template <typename coder_T, typename stream_T, typename source_T>
 Encoder<coder_T, stream_T, source_T>::Encoder(const SymbolStatistics& stats,
                                               size_t probabilityBits) : mSymbolTable(nullptr), mProbabilityBits(probabilityBits)
 {
+  RANSTimer t;
+  t.start();
   mSymbolTable = std::make_unique<encoderSymbolTable_t>(stats, probabilityBits);
+  t.stop();
+  LOG(debug) << "Encoder SymbolTable inclusive time (ms): " << t.getDurationMS();
 }
 
 template <typename coder_T, typename stream_T, typename source_T>
 template <typename stream_IT, typename source_IT>
 const stream_IT Encoder<coder_T, stream_T, source_T>::Encoder::process(
   const stream_IT outputBegin, const stream_IT outputEnd, const source_IT inputBegin, const source_IT inputEnd) const
 {
+  LOG(trace) << "start encoding";
+  RANSTimer t;
+  t.start();
+
   static_assert(std::is_same<typename std::iterator_traits<source_IT>::value_type, source_T>::value);
   static_assert(std::is_same<typename std::iterator_traits<stream_IT>::value_type, stream_T>::value);
 
+  if (inputBegin == inputEnd) {
+    LOG(warning) << "passed empty message to encoder, skip encoding";
+    return outputEnd;
```

Comment:
Why is this a warning? I think an empty input is a totally valid use case. I'd make this a debug at most, in order not to flood the logs.

Reply:
Well, I don't agree on the nullptr thing.
If you have a pair `void* ptr`, `size_t size = 0`, then dereferencing the pointer is by definition invalid, since you will always exceed the size. So an input with size=0 and ptr totally arbitrary is legit.

I agree that it shouldn't and will not be the normal use case to send empty data. But still, there is nothing wrong with it. You could envision something that is not part of every time frame, or sparse data. In my case it was just an event with no track found. So I don't see why this should be a warning. I'd reduce warnings and error to things that need attention.

Reply:
@davidrohr : thanks for explaining. From your side, you are indeed providing legitimate input data. 
On the other hand nullptr as an input iterator or empty messages passed to an entropy coding library in my opinion don't qualify as valid input.
The key point here is that there is a middleware between tracker and entropy coding library which are the entropy coding workflows. Those should check for empty messages and represent them accordingly. As @shahor02 pointed out this is already happening.

Reply:
OK, whatever, if the workflow checks for that already, I don't care. But then we have the check twice...